src/js/components/service-requests/images.js

```diff
@@ -0,0 +1,138 @@
+import { v4 as uuidv4 } from "uuid";
+import { getLabel } from "../../utils/element-factory";
+
+/*
+  We keep a mapping of uploaded images separate from the FileList stored on
+  the input object because we want previously selected images to persist after
+  the user clicks the plus action a second time to upload more images and we want
+  the user to be able to remove images by clicking the cross on a preview.
+
+  Each time the user selects more files to upload, we append these individual
+  File objects to our own uploadedFiles object.
+
+  We later transform this uploadedFiles object into a FormData object
+  instead of submitting the data from the form containing the file input.
+
+  More reading: https://developer.mozilla.org/en-US/docs/Web/API/File/Using_files_from_web_applications
+*/
+
+export function createImageFormFields() {
+  const $uploadImagesLabel = getLabel("Images of your issue");
+  const $formInputTmpl = $(".components .form__input-field:eq(0)");
+  const $uploadImagesInput = $formInputTmpl.clone().attr({
+    id: "upload-images-input",
+    name: "files",
+    type: "file",
+    accept: "image/*", // Beware that the user can override this in a dropdown
+    multiple: true,
+    style: "display: none", // We're using our own element
+  });
+  const $uploadImagesClass = $(".upload-images");
+  const $uploadImagePreviewTemplate = $(".image-preview");
+  const $uploadImageAdd = $(".button.button--add-image");
+
+  // Reroute the click event from the Add element to the hidden input element
+  $uploadImageAdd.click(function () {
+    $uploadImagesInput.click();
+  });
+
+  return {
+    $uploadImagesInput,
+    $uploadImagesLabel,
+    $uploadImagesClass,
+    $uploadImagePreviewTemplate,
+  };
+}
+
+export function toggleSubmitImagesButton($btn, uploadedFiles) {
+  if (Object.keys(uploadedFiles).length > 0) {
+    $btn.show();
+  } else {
+    $btn.hide();
+  }
```

Comment:
Perhaps something like this?

```suggestion
  Object.keys(uploadedFiles).length ? $btn.show() : $btn.hide();
```

Reply:
Thanks @schalkneethling